### ISSUE TITLE
Infinite running - unknwon steps.

### ISSUE THREAD

**Issue by zaabek** [tensorflow/datasets#1279]
I trying run the example https://www.tensorflow.org/tutorials/text/text_classification_rnn

**Environment information**
* Operating System: Windows 8.1
Python version: 3.6.4
tensorflow-datasets/tfds-nightly version: 1.3.2
tensorflow/tensorflow-gpu/tf-nightly/tf-nightly-gpu version: 2.0.0

**Reproduction instructions**
Just run the below code.
```
from __future__ import absolute_import, division, print_function, unicode_literals

import tensorflow_datasets as tfds
import tensorflow as tf

dataset, info = tfds.load('imdb_reviews/subwords8k', with_info=True, as_supervised=True)

train_dataset, test_dataset = dataset['train'], dataset['test']
encoder = info.features['text'].encoder

BUFFER_SIZE = 10000
BATCH_SIZE = 64

train_dataset = train_dataset.shuffle(BUFFER_SIZE)

train_dataset = train_dataset.padded_batch(BATCH_SIZE, tf.compat.v1.data.get_output_shapes(train_dataset))

test_dataset = test_dataset.padded_batch(BATCH_SIZE, tf.compat.v1.data.get_output_shapes(test_dataset))

model = tf.keras.Sequential([
    tf.keras.layers.Embedding(encoder.vocab_size, 64),
    tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(64)),
    tf.keras.layers.Dense(64, activation='relu'),
    tf.keras.layers.Dense(1, activation='sigmoid')
])

model.compile(loss='binary_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])

history = model.fit(train_dataset, epochs=1,
                    validation_data=test_dataset,
                    validation_steps=1)
```

**Link to logs**
     48/Unknown - 332s 7s/step - loss: 0.6901 - accuracy: 0.5326
     49/Unknown - 339s 7s/step - loss: 0.6895 - accuracy: 0.5348
     50/Unknown - 344s 7s/step - loss: 0.6896 - accuracy: 0.5341
     51/Unknown - 367s 7s/step - loss: 0.6891 - accuracy: 0.5355
     52/Unknown - 372s 7s/step - loss: 0.6889 - accuracy: 0.5358
     53/Unknown - 376s 7s/step - loss: 0.6885 - accuracy: 0.5366
     54/Unknown - 382s 7s/step - loss: 0.6879 - accuracy: 0.5376

**Expected behavior**
I thnik the learning should finished.

**Additional context**

**Comment by cyfra** [comment]
Hey,

I think it is working as intended: the dataset has 25'000 objects. Your batch size is 64.
So you should expect around 391 steps.
Based on your log - you waited only to 54 .. :-)

Let it run for a little bit longer, and it should finish once it reaches 391.
If not - please reopen this bug and we can investigate.
